auth_test.go

```diff
@@ -246,6 +247,206 @@ func TestIntegrationList(t *testing.T) {
 	assert.Equal(t, `["dev","github","custom123"]`+"\n", string(b))
 }
 
+type testAuthErrorHTTPHandler struct {
+	wasCalled    bool
+	statusCode   int
+	contentType  string
+	responseBody string
+}
+
+func (h *testAuthErrorHTTPHandler) ServeAuthError(
+	w http.ResponseWriter,
+	_ *http.Request,
+	authError error,
+	reason string,
+	statusCode int,
+) {
+	h.wasCalled = true
+	w.Header().Set("Content-Type", h.contentType)
+	w.WriteHeader(h.statusCode)
+	fmt.Fprint(w, h.responseBody)
+}
+
+func TestIntegrationAuthErrorHTTPHandler(t *testing.T) {
```

Comment:
This would be way more compact and readable if done as a table test. E.g. you need to create a table of `testAuthErrorHTTPHandler` and expected outcomes and test them in a loop with same setup and same checks.

Reply:
I have removed some duplication from this test and described different API calls as a table.